### ISSUE TITLE
build-storybook: Failed to resolve module specifier "vue"

### ISSUE THREAD

**Issue by peteromano** [storybook-vue/nuxt#32]
After building for production, we get this error:

`TypeError: Failed to resolve module specifier "vue". Relative references must start with either "/", "./", or "../".`

![image](https://github.com/storybook-vue/nuxt/assets/392387/ea77cfde-79b0-4a52-a609-0c5f92be2dec)

**Comment by chakAs3** [comment]
hi @peteromano please try to use the latest version

**Comment by peteromano** [comment]
Ok, the versioning was confusing.. I was 0.0.1-rc.29, now on -beta.21, which solves alot of my issues, however, this issue is still happening, and also the _build_ now has an issue (althougyh still works) with: 

![image](https://github.com/storybook-vue/storybook-nuxt/assets/392387/be3979c6-0cfd-4a4c-8e10-c22b1a671289)


My command (i have to run nuxt build again, after it build public/storybook):
 
`"build:testing": "pnpm build-storybook -o public/storybook --disable-telemetry && nuxt build",`

**Comment by chakAs3** [comment]
Hi @th1nkgr33n please  use the latest version of @storybook-vue/nuxt. it will be ok

**Comment by chakAs3** [comment]
> @chakAs3 will try as soon as i can, thanks! BTW, is there a discord we can join to collab?

you can find me on Storybook Discord.  chak-As3

**Comment by th1nkgr33n** [comment]
@chakAs3 thanks for the response.
I switched to the other package "@storybook-vue/nuxt": "0.0.1-beta.22" as you mentioned and adapt the config files. 
But still got the same error.

Do you need further information to investigate?

By the way.. whats the difference between "@storybook-vue/nuxt" and "@storybook-nuxt/framework".
Both namespaces link to the same repo.

**Comment by chakAs3** [comment]
> By the way.. whats the difference between "@storybook-vue/nuxt" and "@storybook-nuxt/framework".

yes exactly i'm just trying name spaces not sure with what i will go .

 

> I switched to the other package "@storybook-vue/nuxt": "0.0.1-beta.22"

i release 0.0.3. yesterday 

you can try cli as well `npx storybook-nuxt init`

**Comment by th1nkgr33n** [comment]
Ah I haven't seen the 0.0.3 release tag in GitHub thats why I thought the beta.22 was latest. I'm not familiar with the process of distributing npm packages under namespaces. THanks.

Unfortunately with 0.0.3 it didn't work as well. (Cleared node_modules and cache)

**Comment by chakAs3** [comment]
@th1nkgr33n can share your repo it should work, i ll take a look

**Comment by peteromano** [comment]
FYI, this is still an issue -- would be FANTASTIC to get this working :)

**Comment by chakAs3** [comment]
> @chakAs3 FYI, this is still an issue -- would be FANTASTIC to get this working :)

Yeah @peteromano  i will Fantastic for many Nuxt devs, i'm finalizing the module this weekend and this Monday i will give live demo in Storybook Discord Channel, if you are interested. it will be great to get your feedback there, and set a roadmap

**Comment by peteromano** [comment]
@chakAs3 oh cool! What time?

Btw, I tried new nuxt 3.7 today in general and there are a few dependency related issues logged in nuxt.. If you see any new issues during your testing, it could be because they need to patch 3.7

3.6.5 still seems most stable

**Comment by simkuns** [comment]
- One way to solve this is to create a file:
`./.storybook/preview-head.html`
```
<script type="importmap">
  {
    "imports": {
      "vue": "https://unpkg.com/vue@3/dist/vue.esm-browser.js"
    }
  }
</script>
```

---

- Another way is to remove `"vue"` from `build.rollupOptions.external` so it gets bundled: (https://github.com/vitejs/vite/discussions/13536)
https://github.com/storybook-vue/storybook-nuxt/blob/7c23918dec333f3f05a54798f3e993374872f35f/packages/storybook-nuxt/src/preset.ts#L138

---

- Or by applying a patch which does the previously mentioned:
```
npx patch-package
```

`./patches/@storybook-vue+nuxt+0.1.0-rc.10.patch`
```
diff --git a/node_modules/@storybook-vue/nuxt/dist/preset.cjs b/node_modules/@storybook-vue/nuxt/dist/preset.cjs
index 6592cf7..146b12d 100644
--- a/node_modules/@storybook-vue/nuxt/dist/preset.cjs
+++ b/node_modules/@storybook-vue/nuxt/dist/preset.cjs
@@ -96,7 +96,7 @@ const viteFinal = async (config, options) => {
   };
   const nuxtConfig = await defineNuxtConfig(await getStorybookViteConfig(config, options));
   return vite.mergeConfig(nuxtConfig.viteConfig, {
-    build: { rollupOptions: { external: ["vue", "vue-demi"] } },
+    build: { rollupOptions: { external: ["vue-demi"] } },
     define: {
       __NUXT__: JSON.stringify({ config: nuxtConfig.nuxt.options.runtimeConfig })
     },
diff --git a/node_modules/@storybook-vue/nuxt/dist/preset.mjs b/node_modules/@storybook-vue/nuxt/dist/preset.mjs
index ba6f7c3..31ddb1e 100644
--- a/node_modules/@storybook-vue/nuxt/dist/preset.mjs
+++ b/node_modules/@storybook-vue/nuxt/dist/preset.mjs
@@ -94,7 +94,7 @@ const viteFinal = async (config, options) => {
   };
   const nuxtConfig = await defineNuxtConfig(await getStorybookViteConfig(config, options));
   return mergeConfig(nuxtConfig.viteConfig, {
-    build: { rollupOptions: { external: ["vue", "vue-demi"] } },
+    build: { rollupOptions: { external: ["vue-demi"] } },
     define: {
       __NUXT__: JSON.stringify({ config: nuxtConfig.nuxt.options.runtimeConfig })
     },
```

**Comment by peteromano** [comment]
@simkuns will try this ASAP.

The latter 2 solutions you mention should come from this repo, though, right? (or I guess I could add solution 2 to sb main vite config) 

THANK YOU very much for the help on this!

**Comment by simkuns** [comment]
@peteromano Yes, the latter 2 solutions mentioned should come from this repo.

You can try upgrading to `@storybook-vue/nuxt@0.1.1` which contains mentioned code changes.

**Comment by chakAs3** [comment]
there is new release 0.1.3
